homeassistant/components/dsmr_reader/config_flow.py

```diff
@@ -0,0 +1,47 @@
+"""Config flow to configure DSMR Reader."""
+from __future__ import annotations
+
+from collections.abc import Awaitable
+from typing import Any
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.config_entry_flow import DiscoveryFlowHandler
+
+from .const import DOMAIN
+
+
+async def _async_has_devices(_: HomeAssistant) -> bool:
+    """Always return true as this integration doesn't support any real devices."""
+    return True
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class DsmrReaderFlowHandler(DiscoveryFlowHandler[Awaitable[bool]], domain=DOMAIN):
+    """Handle DSMR Reader config flow. The MQTT step is inherited from the parent class."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Set up the config flow."""
+        super().__init__(DOMAIN, "DSMR Reader", _async_has_devices)
+
+    async def async_step_import(self, _: dict[str, Any] | None) -> FlowResult:
+        """Import from configuration.yaml and create config entry."""
+        # There's no configuration supported for this integration
```

Comment:
I don't see anything here to guard against creating a new entry each restart

```suggestion
        # There's no configuration supported for this integration
        if self._async_current_entries():
            return self.async_abort(reason="single_instance_allowed")        
```

Reply:
Good catch! That did indeed seem to happen when I added the integration in the configuration and restarted HA. Somewhat annoyingly, I'm seeing the discovery suggest the integration again while it has already been imported from configuration.yaml (or maybe it's a racing condition?). But it detects the other instance when you try to install it again and doesn't snow up any more afterwards, so I think that's acceptable.

This has now been added.